cmd/kubeadm/app/images/puller.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package images
+
+import (
+	"fmt"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	utilsexec "k8s.io/utils/exec"
+)
+
+// ImagePuller is a struct that can pull images and hides the implementation (crictl vs docker)
+type ImagePuller struct {
```

Comment:
Why make it private? I'd also argue that the name should not reveal the implementation as this uses either docker or crictl to pull images

Reply:
The initial lower case was a typo, sorry 

However I'm still not fully comfortable about naming of `ImagePull` and `ImagePuller` because those confused me while reading the code. If we are keeping  'ImagePuller' because it's implementation, what about `ImagePull` ->  `ImageSetPuller` or something that makes more clear it's responsibility to coordinate the pull process for a set of images